hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/common/HoodieSparkEngineContext.java

```diff
@@ -229,6 +231,13 @@ public void cancelAllJobs() {
     javaSparkContext.cancelAllJobs();
   }
 
+  @Override
+  public <I, O> O aggregate(HoodieData<I> data, O zeroValue, Functions.Function2<O, I, O> seqOp, Functions.Function2<O, O, O> combOp) {
+    Function2<O, I, O> seqOpFunc = seqOp::apply;
+    Function2<O, O, O> combOpFunc = combOp::apply;
+    return HoodieJavaRDD.getJavaRDD(data).aggregate(zeroValue, seqOpFunc, combOpFunc);
```

Comment:
This is based on https://github.com/apache/spark/blob/7e8b60b5ae7d6453bc1ce51b5112c975f9aa8757/core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala#L426

Reply:
I didn't see changes for `HoodieMetadataMergedLogRecordScanner` that switches from string cache key to serializable, then how we support the non-string secondary index fields?

Reply:
Please check latest commit. I have done the changes and added a test where we create secondary index for a field of long type.

Reply:
So you are saying the cache key could be string anyway, it's the merger that plays the role for the secondary index?